src/sonic-bgpcfgd/tests/test_bbr.py

```diff
@@ -0,0 +1,319 @@
+from bgpcfgd.directory import Directory
+from bgpcfgd.template import TemplateFabric
+from mock import MagicMock, patch
+from copy import deepcopy
+import swsscommon_test
+import bgpcfgd
+
+with patch.dict("sys.modules", swsscommon=swsscommon_test):
+    from bgpcfgd.managers_bbr import BBRMgr
+
+global_constants = {
+    "bgp": {
+        "allow_list": {
+            "enabled": True,
+            "default_pl_rules": {
+                "v4": [ "deny 0.0.0.0/0 le 17" ],
+                "v6": [
+                    "deny 0::/0 le 59",
+                    "deny 0::/0 ge 65"
+                ]
+            }
+        }
+    }
+}
+
+#@patch('bgpcfgd.managers_bbr.log_info')
+#@patch('bgpcfgd.managers_bbr.log_err')
+#@patch('bgpcfgd.managers_bbr.log_crit')
+def test_constructor():#m1, m2, m3):
+    cfg_mgr = MagicMock()
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr':   cfg_mgr,
+        'tf':        TemplateFabric(),
+        'constants': {},
+    }
+    m = BBRMgr(common_objs, "CONFIG_DB", "BGP_BBR")
+    assert not m.enabled
+    assert len(m.bbr_enabled_pgs) == 0
+    assert m.directory.get("CONFIG_DB", "BGP_BBR", "status") == "disabled"
+
+@patch('bgpcfgd.managers_bbr.log_info')
+@patch('bgpcfgd.managers_bbr.log_crit')
+def set_handler_common(key, value,
+                       is_enabled, is_valid, has_no_push_cmd_errors,
+                       mocked_log_crit, mocked_log_info):
+    cfg_mgr = MagicMock()
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr': cfg_mgr,
+        'tf': TemplateFabric(),
+        'constants': global_constants,
+    }
+    m = BBRMgr(common_objs, "CONFIG_DB", "BGP_BBR")
+    m.enabled = is_enabled
+    prepare_config_return_value = [
+                               ["vtysh", "-c", "clear bgp peer-group PEER_V4 soft in"],
+                               ["vtysh", "-c", "clear bgp peer-group PEER_V6 soft in"]
```

Comment:
in the description, when you enable/disable bbr, you only modify the peer group configuration for the peer group for T0. However, here you seems restart PEER_V4/V6 which is the peer group for T0 as well. Do we need to restart all peer groups, or only T0 peer groups?

Reply:
In the public we have PEER_V4 and PEER_V6, They both work for T2, T1, and T0 peers. So here I'll restart them as T0. There is no way to distinct T1 from T0 (no neighbor meta in the public enabled by requests from community). But internal configs has PEERs which a designed to be used only for T0. In our internal case we restart only required peer groups.